model-optimizer/extensions/middle/UpsampleToResample.py

```diff
@@ -63,6 +64,8 @@ def replace_pattern(self, graph: Graph, match: Dict[str, Node]):
             return
 
```

Comment:
Please fix pattern for this transformation since we use different approach for one node pattern. See `ConcatOptimization` as an example.
Also, I still do not understand why do we need middle transformation. Can't we create front-transformation that will recalculate required inputs for `Interpolate`?

Reply:
It is old transformation (i.e. this transformation existed before the operation Interpolate-4 was specified), and I am not its author. I rewrote this transformation to generate Interpolate-4. Next, we can get something broken if this transformation will be moved into the front stage.